lib/json_schema/validator.rb

```diff
@@ -18,15 +18,19 @@ def initialize(schema)
       @schema = schema
     end
 
-    def validate(data)
+    def validate(data, fail_fast = false)
       @errors = []
       @visits = {}
-      validate_data(@schema, data, @errors, ['#'])
+      @fail_fast = fail_fast
+      define_singleton_method(:custom_and, method(@fail_fast ? :fast_and : :strict_and))
```

Comment:
Man, it never would have occurred to me to use metaprogramming to make things _faster_, hah.

Okay, I don't mind the use of metaprogramming (with the caveat that it could be reverted to a standard method in the future if future additions make its use too confusing/unclean), but could we do a couple things:

1. Space out the `define_singleton_method` line here and add a comment explaining (1) how this works (i.e., something basic like "redefines `custom_and` to `fast_and` for use in "fail fast" mode), and (2) why the metaprogramming approach was used as to opposed to a standard method (i.e., explain that there's a small performance benefit).
2. And just to make this really explicit, could we rename `custom_and` to `strict_and_or_fail_fast`? I'm not super attached to the latter name, but when reading this patch initially I was confused by the naming `custom_and`, and I'd prefer to change it to something a little more descriptive.